api/jsonapi.go

```diff
@@ -4240,47 +4240,11 @@ func (i *jsonAPIHandler) POSTResendOrderMessage(w http.ResponseWriter, r *http.R
 		return
 	}
 
-	if args.OrderID == "" {
-		ErrorResponse(w, http.StatusBadRequest, core.ErrOrderNotFound.Error())
-		return
-	}
-
-	var msgType int32
-	var ok bool
-
-	if msgType, ok = pb.Message_MessageType_value[args.MessageType]; !ok {
-		ErrorResponse(w, http.StatusBadRequest, "invalid order message type")
-		return
-	}
-
-	msg, peerID, err := i.node.Datastore.Messages().
-		GetByOrderIDType(args.OrderID, pb.Message_MessageType(msgType))
-	if err != nil || msg == nil || msg.Msg.GetPayload() == nil {
-		ErrorResponse(w, http.StatusBadRequest, "order message not found")
-		return
-	}
-
-	p, err := peer.IDB58Decode(peerID)
-	if err != nil {
-		ErrorResponse(w, http.StatusBadRequest, "invalid peer id")
+	if err := i.node.ResendCachedOrderMessage(args.OrderID, args.MessageType); err != nil {
```

Comment:
`args.MessageType` is name string. Something like:
```
	var msgType int32
	var ok bool

	if msgType, ok = pb.Message_MessageType_value[args.MessageType]; !ok {
		ErrorResponse(w, http.StatusBadRequest, "invalid order message type")
		return
	}
```
 is required before the call to `ResendCachedOrderMessage` is made.